ansible/roles/ci-scratch-storage/tasks/freebsd.yml

```diff
@@ -0,0 +1,60 @@
+---
+- name: Check vtbd1 disklabel
+  command: gpart list vtbd1
+  ignore_errors: yes
+  register: vtbd1_exists
+
+- name: Create vtbd1 disklabel
+  command: gpart create -s GPT vtbd1
+  become: yes
+  when: vtbd1_exists.rc == 1
+
+- name: Check vtbd1p1 partition
+  command: test -c /dev/vtbd1p1
+  ignore_errors: yes
+  register: vtbd1p1_exists
+
+- name: Create vtbd1p1 partition
+  command: gpart add -a 4k -t freebsd-zfs -l scratch0 vtbd1
+  become: yes
+  when: vtbd1p1_exists.rc == 1
+
+- name: Set up ZFS at boot
+  lineinfile:
+    dest: /etc/rc.conf
+    regexp: '^zfs_enable'
+    line: 'zfs_enable="YES"'
+  become: yes
+  register: rc_zfs
+
+- name: Reload zfs
+  service:
+    name: zfs
+    state: reloaded
+  become: yes
+  when: rc_zfs.changed
```

Comment:
Looking at this, the strict ordering requirements make this trick to do with notify.  I've used this elsewhere.